pyro/poutine/trace_struct.py

```diff
@@ -146,7 +146,7 @@ def log_prob_sum(self, site_filter=lambda name, site: True):
                     if is_validation_enabled():
                         warn_if_nan(log_p, "log_prob_sum at site '{}'".format(name))
                         warn_if_inf(log_p, "log_prob_sum at site '{}'".format(name), allow_neginf=True)
-                result += log_p
+                result = result + log_p
```

Comment:
This is probably the single most traversed line in the entire codebase, so we should make sure there's no performance impact.  cc @neerajprad

Reply:
there also may be some reason this is in-place here. @fritzo is always grumbling about in-place ops, has seen these lines of code many times, so maybe there was some particular reason 🤔

Reply:
Since `log_p` is scalar, perf impact shouldn't be a concern (we can check the hmc test runtimes). I think this should be safe, but it would be best to get a green light from @fritzo.

Reply:
I think it is a good idea to change this, and I don't foresee a performance impact.